File: Maths/509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        f1=0 #initial term
        f2=1 #term 2
        cnt=1 #cnt=1
        if (n==0):
            return f1
        if (n==1):
            return f2
        else:
            while cnt<n:
                fn=f1+f2 #next term is sum of first 2 terms
                f1=f2    #exchanging f1,f2 and fn
                f2=fn
                cnt+=1
        return fn
                
           
        
        

File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        //base case
        if(n==1 || n==0) {
            return n;
        }
        
        int re1=fib(n-1);
        int re2=fib(n-2);
        
        return re1+re2;
    }
};